client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -309,12 +325,26 @@ function StockDefineLotsModalController(
     });
 
     if (vm.errors.length === 0) {
-      saveSetting();
-      Instance.close({
-        lots : vm.form.rows,
-        unit_cost : vm.stockLine.unit_cost,
-        quantity : vm.form.total(),
-      });
+
+      // Maybe update some lot expiration dates
+      const promises = [];
+      if (vm.editExpirationDates) {
+        vm.form.rows.forEach((row) => {
+          const existingLot = getExistingLot(row.lot);
+          if (existingLot && (row.expiration_date !== existingLot.expiration_date)) {
+            promises.push(Lots.update(existingLot.uuid, { expiration_date : row.expiration_date }));
```

Comment:
Are these individual HTTP requests?  It might make sense to just do this when we submit data to the server, rather than try to individually send HTTP requests for each lot.

I think this is probably a performance optimization for later though.  We can merge this in and maybe make improvements if we feel the need to later.